pkg/kv/kvserver/replica_lifecycle_datadriven_test.go

```diff
@@ -426,6 +481,16 @@ func (r *replicaInfo) initialized() bool {
 	return r.hs.Commit > 0 // NB: or r.ts.Index > 0
 }
 
+// pendingSplit represents a split that has been evaluated but not yet applied.
+type pendingSplit struct {
+	trigger *roachpb.SplitTrigger
+	batch   storage.Batch
+}
+
+func (ps *pendingSplit) cleanup() {
+	ps.batch.Close()
+}
```

Comment:
With the above proposal where `batch` is an encoded batch, we won't need the `cleanup()` method.